rover-server/internal/server/server.go

```diff
@@ -0,0 +1,145 @@
+	GetApplicationsInfo(ctx context.Context, params api.GetApplicationsInfoParams) (api.RoverInfoResponse, error)
+	ResetRoverSecret(ctx context.Context, resourceId string) (api.RoverSecretResponse, error)
+}
+
+type ApiSpecificationController interface {
+	Create(ctx context.Context, req api.ApiSpecificationCreateRequest) (api.ApiSpecificationResponse, error)
+	Get(ctx context.Context, resourceId string) (api.ApiSpecificationResponse, error)
+	GetAll(ctx context.Context, params api.GetAllApiSpecificationsParams) (*api.ApiSpecificationListResponse, error)
+	Update(ctx context.Context, resourceId string, req api.ApiSpecificationUpdateRequest) (api.ApiSpecificationResponse, error)
+	Delete(ctx context.Context, resourceId string) error
+	GetStatus(ctx context.Context, resourceId string) (api.ResourceStatusResponse, error)
+}
+
+type EventSpecificationController interface {
+	Create(ctx context.Context, req api.EventSpecificationCreateRequest) (api.EventSpecificationResponse, error)
+	Get(ctx context.Context, resourceId string) (api.EventSpecificationResponse, error)
+	GetAll(ctx context.Context, params api.GetAllEventSpecificationsParams) (*api.EventSpecificationListResponse, error)
+	Update(ctx context.Context, resourceId string, req api.EventSpecificationUpdateRequest) (api.EventSpecificationResponse, error)
+	Delete(ctx context.Context, resourceId string) error
+	GetStatus(ctx context.Context, resourceId string) (api.ResourceStatusResponse, error)
+}
+
+var securityTemplates = map[security.ClientType]security.ComparisonTemplates{
+	security.ClientTypeTeam: {
+		ExpectedTemplate:  "{{ .B.Environment }}--{{ .B.Group }}--{{ .B.Team }}--",
+		UserInputTemplate: "{{ .B.Environment }}--{{ .P.Resourceid }}",
+		MatchType:         security.MatchTypePrefix,
+	},
+	security.ClientTypeGroup: {
+		ExpectedTemplate:  "{{ .B.Environment }}--{{ .B.Group }}--",
+		UserInputTemplate: "{{ .B.Environment }}--{{ .P.Resourceid }}",
+		MatchType:         security.MatchTypePrefix,
+	},
+	security.ClientTypeAdmin: {
+		ExpectedTemplate:  "{{ .B.Environment }}--",
+		UserInputTemplate: "{{ .B.Environment }}--{{ .P.Resourceid }}",
+		MatchType:         security.MatchTypePrefix,
+	},
+}
+
+type Server struct {
+	Log                 logr.Logger
+	ApiSpecifications   ApiSpecificationController
+	Rovers              RoverController
+	EventSpecifications EventSpecificationController
+}
+
+func (s *Server) RegisterRoutes(router fiber.Router) {
+	checkAccess := security.ConfigureSecurity(router, security.SecurityOpts{
+		Enabled: true,
+		Log:     s.Log,
+		JWTOpts: []security.Option[*security.JWTOpts]{
+			security.WithLmsCheck(""),
+			security.WithTrustedIssuers(nil),
+		},
+		BusinessContextOpts: []security.Option[*security.BusinessContextOpts]{
+			security.WithDefaultScope("tardis:team:all"),
+			security.WithLog(s.Log),
+			security.WithScopePrefix("tardis"),
```

Comment:
Should these values remain hard-coded? Making them configurable in the future, would be nice. But might be out-of-scope of this PR